webapp/src/main/java/org/opentestsystem/rdw/admin/web/GroupController.java

```diff
@@ -9,21 +10,26 @@
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import java.util.List;
+import java.util.Set;
 
 @RestController
 public class GroupController {
 
     private final GroupService service;
 
     @Autowired
-    GroupController(GroupService service) {
+    GroupController(final GroupService service) {
         this.service = service;
     }
 
     @GetMapping("/api/groups")
-    public List<Group> get(@AuthenticationPrincipal final User user, GroupQuery query) {
+    public Set<Group> get(@AuthenticationPrincipal final User user, final GroupQuery query) {
         return service.getByQuery(user, query);
     }
 
+    @GetMapping("/api/groups/filters")
+    public GroupFilterOptions get(@AuthenticationPrincipal final User user) {
+        return service.getFilterOptions(user);
```

Comment:
I put this on the same controller as `api/groups?...` as it informs the the api/groups route on what can be queried on.